### PR TITLE
Set active state on list items after refresh

### LINKED ISSUE #341: Active state on object list item not set after page reload
## Describe the bug
Active state on object list item not set after page reload for the following controllers:

- [x] Notifications
- [x] Downtimes
- [x] Comments
- [x] Users
- [x] User/Host/Service Groups
- [x] History

### PR DESCRIPTION
**Pull request by sukhwinder33445**
fix #341

### CODE REVIEW

File: library/Icingadb/Common/BaseTableRowItem.php [hunk 1]
```diff
@@ -79,6 +81,20 @@ protected function createVisual()
...
+
+        return $this;
+    }
+
+    protected function setDetailFilter(Rule $filter)
+    {
+        $this->addAttributes(['data-icinga-detail-filter' => '(' . QueryString::render($filter) . ')']);
+
+        return $this;
+    }
```
Comment (nilmerg) [inline library/Icingadb/Common/BaseTableRowItem.php:96]: The list already provides this with `$this->list->addDetailFilterAttribute()` etc.

File: library/Icingadb/Common/BaseListItem.php [hunk 2]
```diff
@@ -157,7 +157,7 @@ protected function setMultiselectFilter(Rule $filter)
 
     protected function setDetailFilter(Rule $filter)
     {
-        $this->addAttributes(['data-icinga-detail-filter' => QueryString::render($filter)]);
+        $this->addAttributes(['data-icinga-detail-filter' => '(' . QueryString::render($filter) . ')']);
```
Comment (nilmerg) [inline library/Icingadb/Common/BaseListItem.php:160]: There is `Icinga\Module\Icingadb\Common\DetailActions` for this actually. Any user of this method and the one above should use the list's utility methods instead.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Icingadb/Common/BaseListItem.php (hunk 2 above)
Comment (sukhwinder33445) [reply library/Icingadb/Common/BaseListItem.php:160]: Okay, I will add `DetailActions` instead. 
The parentheses were missing anyway or not?


> should use the list's utility methods instead

You mean the `DetailActions` ?

Comment (nilmerg) [reply library/Icingadb/Common/BaseListItem.php:160]: Take a look into e.g. `BaseHostListItem` on how this handles this. Then I guess you'll know what I mean?

**Review by nilmerg — CHANGES_REQUESTED** [review]
Please also remove the now unused methods.

**Comment by sukhwinder33445** [comment]
I have already removed the methods from `BaseTableRowItem`.  Or do you mean the method in the class `BaseListItem`? 🤔

**Review by nilmerg — APPROVED** [review]
> Or do you mean the method in the class BaseListItem?

Exactly those :)

**Pull request merged**